hell/adventcalendar/2025/16/index.md

```diff
@@ -1,35 +1,114 @@
 ---
-title: "Something something about the Invoker Commands API"
-author: "Your Name"
-author_bio: "Your short bio"
+title: "Controlling dialogs and popovers with the Invoker Commands API"
+author: "Aubrey Sambor"
+author_bio: "Aubrey Sambor is a front-end developer and accessibility advocate. She's one of the organizers of [A11yTalks](https://a11ytalks.com), a monthly virtual meetup featuring speakers and conversations around digital accessibility. She loves CSS, fountain pens, knitting, coffee, and sushi."
 date: 2025-12-16
 author_links:
-  - label: "Site"
-    url: "https://linktoyourblog123.com"
-    link_label: "linktoyourblog123.com"
+  - label: "Aubrey's website"
+    url: "https://aubreysambor.com"
+    link_label: "aubreysambor.com"
+  - label: "Aubrey on Mastodon"
+    url: "https://labyrinth.social/@starshaped"
+    link_label: "@starshaped@labyrinth.social"
 intro: "<p>Short introductory text</p>"
 image: "advent25_16"
 ---
-Some text.
-Some text.
+The [Invoker Commands API](https://developer.mozilla.org/en-US/docs/Web/API/Invoker_Commands_API) is a new way to control popovers and modal dialogs without having to write JavaScript. Dialogs have been available in all modern browsers since March 2022, and the Popover API is available in all modern browsers as of January 2025.
 
-Some text. Some text.
+Until now, users wanting to implement the `<dialog>` element needed to write their own JavaScript to power the show and hide functionality using the `HTMLDialogElement` interface, while the Popover API and Invoker Commands API for popovers work identically by using `HTMLElement` attributes to show, hide, or toggle the popover.
 
-## Heading
+Why should you use this new API, and what benefits does it bring?
 
-Some text.
+## Dialogs vs popovers
+First, let's go over the difference between a dialog and a popover. Hidde de Vries [wrote a blog post in 2022](https://hidde.blog/dialog-modal-popover-differences/) detailing the differences, but in short:
 
-## Heading
+* Dialogs usually contain an action to take, such as agreeing to text or choosing an option. Popovers usually display short-lived information such as a date picker or toast notification.
+* A dialog must be explicitly closed, either by taking an action within the dialog or closing via a close button, while a popover can be lightly dismissed by clicking outside of the popover.
+* A dialog can either be modal or non-modal, while a popover is always non-modal by design. As of late 2025 the Invoker Commands API only has a built-in command for a modal dialog, so I will be focusing on modal dialogs only in this post.
+* Dialogs have a built in backdrop functionality to tint the background a different color, while popovers shouldn't use a backdrop in most cases.
+* A dialog is an HTML element with a role of `dialog`, while a popover is considered an attribute and does not come with a role.
```

Comment:
Yes and no. It doesn't come with a role but if you put popover on a generic element like div, Chromium-based browsers and Firefox turn the generic div into a group, as if you where using role=group explicitly.